cmd/connect.go

```diff
@@ -72,11 +74,28 @@ func (o *ConnectOptions) Complete(f Factory, args []string) (err error) {
 			return err
 		}
 	} else if !QRIRepoInitialized(qriPath) {
-		return errors.New(repo.ErrNoRepo, "no qri repo exists\nhave you run 'qri setup'?")
+		return qerr.New(repo.ErrNoRepo, "no qri repo exists\nhave you run 'qri setup'?")
 	}
 
 	if err = f.Init(); err != nil {
-		return err
+		if errors.Is(err, qipfs.ErrNeedMigration) {
```

Comment:
We need to migrate before this point, I am getting a `nil pointer dereference` error before we even get into the connect command code:

In our cmd `PersistentPreRun` func, we call `opt.Instance()` to get an instance so we can get to the config. `opt.Instance()` runs `o.Init()`, which calls `lib.NewInstance`, which errors with the ipfs `ErrNeedsMigration` error. `o.Init()` returns the error, and `opt.Instance()` does not display the error, but does pass back `nil` as the instance pointer. When we try to access the config using this nil instance pointer, we get the `nil pointer dereference` error. 

Made a change to move the migration into `o.Init()`, handling the confirmation there. But this also feels wrong, and seems like it should happen when we try to give the Instance the `Filestore`, but still feeling for what is appropriate.

However, making this change meant I successfully migrated my repo!

Reply:
made these changes on branch `chore_ipfs_bump_ramfox`, didn't want to push to this branch before knowing if these changes made sense